Tools & Armour category in JEI

uecasm opened the pull request:
# Changes proposed in this pull request:
- Adds a page to JEI that shows the tools & armour colonists will accept at each hut level.
![image](https://user-images.githubusercontent.com/539951/236663908-b03f1659-30ce-477c-876d-a29a513f512f.png)
![image](https://user-images.githubusercontent.com/539951/236663912-c1df7cce-2127-4c1e-b835-265a4dd4bd8d.png)
![image](https://user-images.githubusercontent.com/539951/236663914-52d5e35b-37c8-4b84-80fe-ee691b03207e.png)
- The first row is base tools, the second row are tools with some example enchants of each level (not comprehensive, just illustrative).
- As best I can tell, this correctly replicates the logic that the mod itself uses to decide if tools are the right level or not.  Interestingly, this differs from what the wiki and discord claim about how enchantments are treated.
    - In particular: any number of level 1 enchants can be applied without increasing the tool level; any number of level 2 enchants increase the level by 1 (in total, not each), etc.

Review please (could port)

uecasm committed:
Tools & Armour category in JEI

marchermans reviewed (dismissed):
Very nice work!  Looks amazing!

uecasm committed:
Tool audit file

uecasm commented:
Added audit file and updated [the spreadsheet](https://docs.google.com/spreadsheets/d/1DaGFlFcnce4bHXHylzztr5_Rwnm3P42c6o_xXqgyyr8/edit#gid=1551679170) to show all base tools.
